homeassistant/components/websocket_api/connection.py

```diff
@@ -60,6 +66,28 @@ def context(self, msg: dict[str, Any]) -> Context:
         """Return a context."""
         return Context(user_id=self.user.id)
 
+    @callback
+    def async_register_binary_handler(
+        self, handler: BinaryHandler
+    ) -> tuple[int, Callable[[], None]]:
+        """Register a temporary binary handler for this connection.
+
+        Returns a binary prefix (1 byte) and a callback to unregister the handler.
+        """
+        prefix = 1
+        while prefix in self.binary_handlers and prefix < 256:
+            prefix += 1
```

Comment:
This could be a list instead of a `dict` since position in the list will always be known